src/layman/upgrade/upgrade_v1_21_test.py

```diff
@@ -79,3 +82,32 @@ def test_wfs_wms_status(publ_type, publ_name, rest_args, exp_wfs_wms_status):
         assert wfs_wms_status is None, f'publication={publ_type}:{workspace}.{publ_name}, wfs_wms_status={wfs_wms_status}'
 
     process_client.delete_workspace_publication(publ_type, workspace, publ_name)
+
+
+@pytest.mark.usefixtures('ensure_layman')
+def test_adjust_layer_metadata_url_on_gs():
+    workspace = 'test_adjust_metadata_url_workspace'
+
+    layers = [
+        ('vector_sld', common_publications.LAYER_VECTOR_SLD.definition),
+        ('vector_qml', common_publications.LAYER_VECTOR_QML.definition),
+        ('raster', common_publications.LAYER_RASTER.definition),
+    ]
+
+    for layer, layer_def in layers:
+        process_client.publish_workspace_publication(process_client.LAYER_TYPE, workspace, layer, **layer_def)
+
+    for layer, _ in layers:
+        asserts_util.is_publication_valid_and_complete(Publication(workspace=workspace,
+                                                                   type=process_client.LAYER_TYPE,
+                                                                   name=layer))
```

Comment:
Remove metadataLinks, check MetadataURL is not in WMS/WFS capabilities and then run migration.